src/BinaryParsers/PEBinary/PortableExecutable/PE.cs

```diff
@@ -889,13 +888,58 @@ private ChecksumAlgorithmType ChecksumAlgorithmForFullPdb(Pdb pdb)
                 foreach (DisposableEnumerableView<SourceFile> sfView in pdb.CreateSourceFileIterator(om))
                 {
                     SourceFile sf = sfView.Value;
-                    return sf.HashType == HashType.SHA256 ? ChecksumAlgorithmType.Sha256 : ChecksumAlgorithmType.Sha1;
+                    return GetChecksumAlgorithmType(sf.HashType);
                 }
             }
 
             return ChecksumAlgorithmType.Unknown;
         }
 
+        private ChecksumAlgorithmType GetChecksumAlgorithmType(HashType hashType)
+        {
+            ChecksumAlgorithmType checksumAlgorithmType;
+
+            switch (hashType)
+            {
+                case HashType.SHA256:
+                    checksumAlgorithmType = ChecksumAlgorithmType.Sha256;
+                    break;
+                case HashType.MD5:
+                    checksumAlgorithmType = ChecksumAlgorithmType.Md5;
+                    break;
+                case HashType.SHA1:
+                    checksumAlgorithmType = ChecksumAlgorithmType.Sha1;
+                    break;
+                case HashType.None:
+                    checksumAlgorithmType = ChecksumAlgorithmType.Unknown;

```

Comment:
>checksumAlgorithmType [](http://example.com/codeflow?start=20&length=21)

So what happens in this code path? Would we report an error? Is this an expected condition or is this another exceptional code path?

Reply:
The consumer of the code that gets the checksumAlgorithmType back will see that the type is NOT SHA256 therefore it will still be an error. It just affects what will be display. There is no further logic that tests the value (it is only if (hashFound != SHA256) then generate error and the message in this case would contain the word 'Unknown' for the the hash type found instead of 'Sha-1' or 'Md5'. So no code flow difference just a messaging change.